app/lib/goalmodeler/Modeler.js

```diff
@@ -138,14 +138,14 @@ Modeler.prototype.createObject = function (name) {
   return shape.businessObject;
 }
 
-Modeler.prototype.renameObject = function (clazz, name) {
-  this.get('modeling').updateLabel(this.get('elementRegistry').get(clazz.id), name);
+Modeler.prototype.renameObject = function (object, name) {
+  this.get('modeling').updateLabel(this.get('elementRegistry').get(object.id), name);
 }
 
-Modeler.prototype.deleteObject = function (clazz) {
-  this.get('modeling').removeShape(this.get('elementRegistry').get(clazz.id));
+Modeler.prototype.deleteObject = function (object) {
+  this.get('modeling').removeShape(object);
```

Comment:
How did you find out, you do not need the step: this.get('elementRegistry').get(object.id)

Reply:
We followed the deletion of an object in the data modeler and in the goal modeler and went through trial and error from there...